src/test/groovy/graphql/execution/preparsed/persisted/ApolloPersistedQuerySupportTest.groovy

```diff
@@ -152,4 +155,18 @@ class ApolloPersistedQuerySupportTest extends Specification {
         then:
         documentEntry.hasErrors()
     }
-}
\ No newline at end of file
+
+    def "will have error if the calculated sha hash of the query does not match the persistedQueryId"() {
+        def cache = new CacheImplementation()
+        def apolloSupport = new ApolloPersistedQuerySupport(cache)
+        when:
+        def ei = mkEI("badHash", PERSISTED_QUERY_MARKER)
+        def docEntry = apolloSupport.getDocument(ei, engineParser)
+        then:
+        docEntry.getDocument() == null
+        def error = docEntry.getErrors()[0]
+        error.message == "PersistedQueryIdInvalid"
```

Comment:
Is this the best error message.  The error classification could put in a type say 

I am always torn between human readable error messages and ones that computers can switch on.

Given this is unexpected I think a human oriented string is better